### PR TITLE
feat(CLI): Automatically expand loaded env variables

### PR DESCRIPTION
**Pull request by d-asensio**
Closes: #9689

### CODE REVIEW

**Comment by pgrzesik** [comment]
Hello @d-asensio - could you please include motivation behind this PR proposal? I believe there is not corresponding issue created for that , correct?

**Comment by d-asensio** [comment]
Hey @pgrzesik sure! I found some time to write an issue with the motivations of this. I referenced it in my first comment.

Thank you!

**Review by pgrzesik — COMMENTED** [review]
Thanks a lot @d-asensio - it looks great overall and thanks so much for writing up a motivation in a linked issue. I think it's a good idea to allow such expansions, but it would also be good to document it e.g. here: https://github.com/serverless/serverless/blob/master/docs/environment-variables.md

Btw - it looks like there's a conflicting file - could you please look into that and rebase your branch?

**Comment by d-asensio** [comment]
Hey, @pgrzesik I rebased it onto master and added a comment with a little example in the docs.

Now the dependencies security check is failing, but I think is due to the rebase, since [in master it is failing as well](https://snyk.io/test/github/serverless/serverless#SNYK-JS-ADMZIP-1065796). In any case, I don't have permission to check snyk for this branch, could you help me with this?

Thank you!

**Comment by d-asensio** [comment]
Hi @pgrzesik, is there something missing here?

File: docs/environment-variables.md [hunk 1]
```diff
@@ -12,6 +12,17 @@ With `useDotenv: true` set in your `serverless.yml` file, framework automaticall
 
 The framework looks for `.env` and `.env.{stage}` files in service directory and then tries to load them using `dotenv`. If `.env.{stage}` is found, `.env` will not be loaded. If stage is not explicitly defined, it defaults to `dev`.
 
+### Variable expansion
```
Comment (pgrzesik) [inline docs/environment-variables.md:15]: Let's also mention what specific library is being used to perform this expansion

**Review by pgrzesik — COMMENTED** [review]
Thanks a lot :bow: Sorry for the delay in review - I missed the notification and then I was off for a few days. I have one minor suggestion related to docs - also, let's wait for opinion from @medikoo if we should introduce such extensions.

**Review by medikoo — APPROVED** [review]
@d-asensio @pgrzesik it feels as great addition to me.  It mimicks how things will work in a shell, which is a nice improvement.

@pgrzesik After adjustment to docs is made, feel free to merge

**Comment by pgrzesik** [comment]
Thanks @medikoo :bow: @d-asensio so the only thing to address is one comment about attributing what library is used for expansions and we're good to go :+1:

**Comment by d-asensio** [comment]
Done! Thank you both @pgrzesik and @medikoo! 🥳 🎉

**Review by pgrzesik — APPROVED** [review]
Thank you @d-asensio, well done :clap:

**Pull request merged**